File: src/main/java/rk/cinema/model/Customer.java
package rk.cinema.model;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import jakarta.validation.constraints.NotBlank;

@MappedEntity
public record Customer(
        @Id
        @GeneratedValue
        Long id,

        @NonNull
        @NotBlank
        String email,

        @NonNull
        @NotBlank
        String name
) {
}

File: src/main/kotlin/rk/cinema/model/Customer.kt
package rk.cinema.model

import io.micronaut.core.annotation.NonNull
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import jakarta.validation.constraints.NotBlank

@MappedEntity
data class Customer(
    @GeneratedValue @param:Id val id: Long,
    val email: @NonNull @NotBlank String,
    val name: @NonNull @NotBlank String,
)
